Source/WebCore/rendering/StyledMarkedText.cpp

```diff
@@ -136,18 +210,38 @@ Vector<StyledMarkedText> StyledMarkedText::subdivideAndResolve(const Vector<Mark
         return { styledMarkedText };
     }
 
-    auto markedTexts = MarkedText::subdivide(textsToSubdivide);
+    auto markedTexts = MarkedText::subdivide(textsToSubdivide, OverlapStrategy::None);
     ASSERT(!markedTexts.isEmpty());
     if (UNLIKELY(markedTexts.isEmpty()))
         return { };
 
+    bool containsHighlights = false;
+
+    if (!markedTexts.isEmpty()) {
+        // Check if vector contains custom highlights.
+        containsHighlights = markedTexts.containsIf([](const auto& item) {
```

Comment:
the variable can just be both declared and defined here, right?

Reply:
yes, it was outside bc I was accessing it outside before, but I changed it now.